Hero.cpp

```diff
@@ -0,0 +1,60 @@
+#include <string>
+#include <iostream>
+#include "Hero.h"
+
+
+Hero::Hero(const std::string name, const int hp, const int dmg) : name(name), hp(hp), dmg(dmg){
+    this->name = name;
+    this->hp = hp;
+    this->dmg = dmg;
+}
+
+std::string Hero::getName(){
+    return name;
+    
+}
+void Hero::setName(std::string name){
+    this->name = name;
+    
+}
+int Hero::getHP(){
+    return hp;
+    
+}
+void Hero::setHP(int hp){
+    this->hp = hp;
+    
+}
+int Hero::getDmg(){
+    return dmg;
+    
+}
+void Hero::setDmg(int dmg){
+    this->dmg = dmg;
+    
+}
+
+bool Hero::isDead(){
+    if(getHP()<=0){
+        return true;
+    }else{
+        return false;
+    }
+    
+}
+
+void Hero::damaging(Hero *enemy){
+    if(hp > 0){
+        std::cout << name << " is damaging "<< enemy->name << " with: " << dmg << " dmg" << std::endl;
```

Comment:
Jobb lenne a kiíratást kivenni a metódusból. Egy metódusnak feladata legyen! (single responsibility)

Reply:
@oliverosz az megoldható, hogy a Hero.cpp-be külön függvényt hozok létre a kiiratásoknak? Vagy azokat célszerűbb a main.cpp-ben megcsinálni külön függvénybe?